### PR TITLE
fix some changes

### PR DESCRIPTION
**Pull request by khalil2535**
lots of files but few changes yet

### CODE REVIEW

File: Compression/src/compressclient.java [hunk 1]
```diff
@@ -1,3 +1,5 @@
+package Compression.src;
+
 
 public class compressclient {
```
Comment (christianbender) [inline Compression/src/compressclient.java]: Can you change the name of the class according to the Java code conventions

File: ciphers/AES.java [hunk 2]
```diff
@@ -3,552 +3,589 @@
...
+
+        // Invert initial round
+        cipherText = addRoundKey(cipherText, roundKeys[0]);
+
+        return cipherText;
+    }
+
+    public static void main(String[] args) {
+
+        try (Scanner input = new Scanner(System.in)) {
```
Comment (christianbender) [inline ciphers/AES.java]: Please change this construct. 
```java
try {
// somehting

}
catch (<Exception> e) {
// somehting
}
```

File: ciphers/AES.java [hunk 3]
```diff
@@ -3,552 +3,589 @@
...
+
+            // add the current RCON value to the first byte
+            if (i == 0) {
+                currentByte = currentByte ^ RCON[rconCounter];
+            }
+
+            currentByteBits = Integer.toHexString(currentByte);
+
+            // Add zero padding
+            while (currentByteBits.length() < 2) {
```
Comment (christianbender) [inline ciphers/AES.java]: Assign the length of the string to a variable. For example ```currentByteBitsLength```. This is more efficient.

File: ciphers/AES.java [hunk 4]
```diff
@@ -3,552 +3,589 @@
...
+    /**
+     *
+     * Returns an array of 10 + 1 round keys that are calculated by using
+     * Rijndael key schedule
+     *
+     * @param initialKey
+     * @return array of 10 + 1 round keys
+     */
+    public static BigInteger[] keyExpansion(BigInteger initialKey) {
+        BigInteger[] roundKeys = {initialKey, new BigInteger("0"), new BigInteger("0"), new BigInteger("0"),
```
Comment (christianbender) [inline ciphers/AES.java]: Split this line. Max. 80 columns per line.

File: ciphers/AES.java [hunk 5]
```diff
@@ -3,552 +3,589 @@
...
+     * @param block of 128-bit integers
+     * @return array of 8-bit integers
+     */
+    public static int[] splitBlockIntoCells(BigInteger block) {
+
+        int[] cells = new int[16];
+        String blockBits = block.toString(2);
+
+        // Append leading 0 for full "128-bit" string
+        while (blockBits.length() < 128) {
```
Comment (christianbender) [inline ciphers/AES.java]: Same thing as above.

File: ciphers/AES.java [hunk 6]
```diff
@@ -3,552 +3,589 @@
...
+        return cipherText;
+    }
+
+    public static void main(String[] args) {
+
+        try (Scanner input = new Scanner(System.in)) {
+            System.out.println("Enter (e) letter for encrpyt or (d) letter for decrypt :");
+            char choice = input.nextLine().charAt(0);
+            String in;
+            if (choice == 'E' || choice == 'e') {
```
Comment (christianbender) [inline ciphers/AES.java]: Here you can use a switch-construct instead.

File: ciphers/AES.java [hunk 7]
```diff
@@ -3,552 +3,589 @@
...
+                System.out.println("Choose a plaintext block (128-Bit Integer in base 16):");
+                in = input.nextLine();
+                BigInteger plaintext = new BigInteger(in, 16);
+                System.out.println("Choose a Key (128-Bit Integer in base 16):");
+                in = input.nextLine();
+                BigInteger key = new BigInteger(in, 16);
+
+                System.out.println("The encrypted message is: \n" + encrypt(plaintext, key).toString(16));
+            }
+            if (choice == 'D' || choice == 'd') {
```
Comment (christianbender) [inline ciphers/AES.java]: Same thing as above.

File: Compression/src/compressclient.java (hunk 1 above)
Comment (khalil2535) [reply Compression/src/compressclient.java]: yep , done

File: ciphers/AES.java (hunk 2 above)
Comment (khalil2535) [reply ciphers/AES.java]: Why ?
this is try block with resources and it's available in Java from JDK8 (Java 8)
this is my references about it from Oracle Java docs 
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
if you still want it a normal try-catch block then I'll do it.

File: ciphers/AES.java (hunk 3 above)
Comment (khalil2535) [reply ciphers/AES.java]: :+1: done here

File: ciphers/AES.java (hunk 5 above)
Comment (khalil2535) [reply ciphers/AES.java]: :+1:

File: ciphers/AES.java (hunk 4 above)
Comment (khalil2535) [reply ciphers/AES.java]: :+1:

File: ciphers/AES.java (hunk 7 above)
Comment (khalil2535) [reply ciphers/AES.java]: :+1:

**Comment by khalil2535** [comment]
in the read me why did you put TheAlgorithms/Python project build ?

**Comment by khalil2535** [comment]
Closed for conflicts

**Pull request closed without merging**